src/components/calcite-picker/calcite-picker.tsx

```diff
@@ -0,0 +1,291 @@
+  //
+  //  Private Methods
+  //
+  // --------------------------------------------------------------------------
+
+  setupRows(): void {
+    this.rows = Array.from(this.el.querySelectorAll("calcite-picker-row"));
+    this.rows.forEach((row) => {
+      row.setAttribute("icon", this.getIconType());
+    });
+    if (this.dragEnabled && this.mode === "configuration") {
+      this.setUpDragAndDrop();
+    }
+  }
+
+  setUpDragAndDrop(): void {
+    const sortGroups = [this.el, ...Array.from(this.el.querySelectorAll("calcite-picker-group"))];
+    sortGroups.forEach((sortGroup) => {
+      Sortable.create(sortGroup, {
+        group: this.el.id,
+        handle: `.${CSS.dragHandle}`,
+        draggable: "calcite-picker-row"
+      });
+    });
+  }
+
+  deselectRow(item: HTMLCalcitePickerRowElement): void {
+    item.removeAttribute("selected");
+    this.selectedValues.delete(item);
+  }
+
+  startEdit(): void {
+    this.editing = true;
+  }
+
+  cancelDelete(): void {
+    this.deletedRows.forEach((row: HTMLCalcitePickerRowElement) => {
+      row.removeAttribute("hidden");
+    });
+    this.deletedRows = new Set();
+    this.editing = false;
+  }
+
+  confirmDelete(): void {
+    let selectedChanged = false;
+    this.deletedRows.forEach((row: HTMLCalcitePickerRowElement) => {
+      if (this.selectedValues.has(row)) {
+        this.selectedValues.delete(row);
+        selectedChanged = true;
+      }
+      row.remove();
+    });
+    if (selectedChanged) {
+      this.calcitePickerSelectionChange.emit(this.selectedValues);
+    }
+    this.calcitePickerRowsDeleted.emit(this.deletedRows);
+    this.deletedRows = new Set();
+    this.editing = false;
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Public Methods
+  //
+  // --------------------------------------------------------------------------
+
+  @Method() async getSelectedRows(): Promise<Set<object>> {
+    return this.selectedValues;
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Render Methods
+  //
+  // --------------------------------------------------------------------------
+
+  getIconType() {
+    let type = null;
+    if (this.mode === "configuration" && this.dragEnabled) {
+      type = "grip";
+    } else if (this.mode === "selection" && this.multiple) {
+      type = "square";
+    } else if (this.mode === "selection" && !this.multiple) {
+      type = "circle";
+    }
+    return type;
+  }
+
+  renderSecondaryAction(action) {
+    return action ? (
+      <calcite-action slot="secondaryAction" onClick={action.onclick || void 0}>
+        <CalciteIcon size={action.icon.size} path={action.icon.path} />
+      </calcite-action>
+    ) : null;
+  }
+
+  renderEditButton() {
+    return this.editEnabled ? (
+      !this.editing ? (
+        <section>
+          <calcite-action onClick={() => this.startEdit()}>
```

Comment:
Add some text to the action for accessibility. may apply to other places

Reply:
edit is going away.